xarray/backends/common.py

```diff
@@ -27,8 +22,41 @@
 NONE_VAR_NAME = '__values__'
 
 
-# dask.utils.SerializableLock if available, otherwise just a threading.Lock
-GLOBAL_LOCK = Lock()
+def get_scheduler(get=None, collection=None):
+    try:
+        from dask.utils import effective_get
+        actual_get = effective_get(get, collection)
+        try:
+            from dask.distributed import Client
+            if isinstance(actual_get.__self__, Client):
+                return 'distributed'
+        except (ImportError, AttributeError):
+            import dask.multiprocessing
```

Comment:
I think we need to guard against this line potentially raising `ImportError` because cloudpickle is missing (which is only required for dask.multiprocessing).